### PR TITLE
addWrapper()

### PR DESCRIPTION
**Pull request by helixbass**
In this PR:
- type `flowMax()` as `typeof flow` (for now)
- mostly-working typing for `addWrapper()`

Based on `addStateHandlers`

### CODE REVIEW

File: types/ad-hok/index.d.ts [hunk 1]
```diff
@@ -63,4 +66,315 @@ declare module 'ad-hok' {
   ) => TProps & TState & { [K in keyof Updaters]: ReturnType<Updaters[K]> }
 
   declare const addStateHandlers: AddStateHandlersType
+
+  type Many<T> = T | ReadonlyArray<T>
+
+  interface LodashFlow {
```
Comment (helixbass) [inline types/ad-hok/index.d.ts]: I tried copying this from the `lodash/fp` types and then just doing `declare const flowMax: LodashFlow`, but that wasn't working

So then the only change I tried was to update `lodash.Many` -> `Many` inside this definition and copy in the definition of `Many` from `lodash` (above)

Didn't dig that far into it, just imported `flow` and made `flowMax` typed as `typeof flow` for now

Comment (peterstuart) [reply types/ad-hok/index.d.ts]: Could we move this into a separate file? It's already pretty big, and once we add a custom version support a much higher arity, it's going to be ridiculous.

File: src/App.tsx [hunk 2]
```diff
@@ -40,6 +43,33 @@ interface AppProps {
...
 }
 
+// interface PropAddingAddWrapperOptions<TProps> {
+//   render: (additionalProps: { val: string }) => any
+//   props: TProps
+// }
+
+const Max: FC = flowMax(
+  addState('num', 'setNum', 0),
+  // addWrapper(({ render, props: { setNum } }: PropAddingAddWrapperOptions) => (
```
[the remark this answers is not in the document]
Comment (peterstuart) [reply src/App.tsx]: In general, my preference would be to get types working for the common case (I wasn't even aware that you could provide additional props to `render`) and not worry too much about stuff like this for now. In this particular case, though, I think the API change is pretty reasonable, and actually matches the API of `addStateHandlers` better (`state` and `props` are positional). 🤷‍♂

[the remark this answers is not in the document]
Comment (helixbass) [reply src/App.tsx]: Yup I'm amenable to the idea that it's not necessarily a worse call signature for `addWrapper()` period

Not that I'd be excited about making a breaking change to the `ad-hok` API but let's keep this in mind

File: types/ad-hok/index.d.ts (hunk 1 above)
Comment (helixbass) [reply types/ad-hok/index.d.ts]: I assume you took care of this in your `flowMax()` PR? But yes agreed

Comment (peterstuart) [reply types/ad-hok/index.d.ts]: Yep, I put it in a separate file.

**Pull request merged**